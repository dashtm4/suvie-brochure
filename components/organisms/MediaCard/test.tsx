import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { MediaCard } from '.'

describe('<MediaCard />', () => {
  it('should renders correct <MediaCard />', () => {
    const title = 'Test title'
    const description = 'Test description'
    const imageSrc = 'image.jpg'

    const wrapper = shallow(
      <MediaCard title={title} description={description} imageSrc={imageSrc} />
    )

    expect(wrapper.contains(title)).toBeTruthy()
    expect(wrapper.contains(description)).toBeTruthy()

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should renders correct <MediaCard /> with correct videoSrc prop', () => {
    const title = 'Test title'
    const description = 'Test description'
    const imageSrc = 'image.jpg'
    const videoSrc = 'video.avi'

    const wrapper = shallow(
      <MediaCard
        title={title}
        description={description}
        imageSrc={imageSrc}
        videoSrc={videoSrc}
      />
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should renders correct <MediaCard /> with correct tags prop', () => {
    const title = 'Test title'
    const description = 'Test description'
    const imageSrc = 'image.jpg'
    const tag1 = 'Test tag 1'
    const tag2 = 'Test tag 2'
    const tags = [tag1, tag2]

    const wrapper = shallow(
      <MediaCard
        title={title}
        description={description}
        imageSrc={imageSrc}
        tags={tags}
      />
    )

    expect(wrapper.contains(tag1)).toBeTruthy()
    expect(wrapper.contains(tag2)).toBeTruthy()

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
