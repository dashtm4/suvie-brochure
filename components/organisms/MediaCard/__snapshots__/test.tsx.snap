// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MediaCard /> should renders correct <MediaCard /> 1`] = `
<div
  className="text-center md:text-left"
>
  <figure
    className="relative mb-3 md:mb-4"
    style={
      Object {
        "height": "420px",
      }
    }
  >
    <div
      className="h-full overflow-hidden relative md:rounded-lg"
    >
      <VideoPlayer
        autoplay={false}
        className="object-cover absolute inset-0 w-full h-full"
        controls={false}
        loop={true}
        muted={true}
        playerRef={
          Object {
            "current": null,
          }
        }
        playsInline={true}
        poster="image.jpg"
      />
    </div>
    <button
      className="absolute inset-0 w-full h-full"
      onClick={[Function]}
      style={
        Object {
          "background": "linear-gradient(0deg, rgba(0, 0, 0, 0.75) 0%, rgba(0, 0, 0, 0.15) 50%)",
        }
      }
    />
  </figure>
  <h3
    className="px-3 mb-2 font-serif text-2xl font-medium md:px-0"
  >
    Test title
  </h3>
  <div
    className="px-3 max-w-sm md:px-0"
  >
    Test description
  </div>
</div>
`;

exports[`<MediaCard /> should renders correct <MediaCard /> with correct tags prop 1`] = `
<div
  className="text-center md:text-left"
>
  <figure
    className="relative mb-3 md:mb-4"
    style={
      Object {
        "height": "420px",
      }
    }
  >
    <div
      className="h-full overflow-hidden relative md:rounded-lg"
    >
      <VideoPlayer
        autoplay={false}
        className="object-cover absolute inset-0 w-full h-full"
        controls={false}
        loop={true}
        muted={true}
        playerRef={
          Object {
            "current": null,
          }
        }
        playsInline={true}
        poster="image.jpg"
      />
    </div>
    <button
      className="absolute inset-0 w-full h-full"
      onClick={[Function]}
      style={
        Object {
          "background": "linear-gradient(0deg, rgba(0, 0, 0, 0.75) 0%, rgba(0, 0, 0, 0.15) 50%)",
        }
      }
    />
    <div
      className="flex absolute left-0 top-2 gap-2 justify-center px-4 w-full md:justify-start md:top-4"
    >
      <Badge
        key="Test tag 1"
      >
        Test tag 1
      </Badge>
      <Badge
        key="Test tag 2"
      >
        Test tag 2
      </Badge>
    </div>
  </figure>
  <h3
    className="px-3 mb-2 font-serif text-2xl font-medium md:px-0"
  >
    Test title
  </h3>
  <div
    className="px-3 max-w-sm md:px-0"
  >
    Test description
  </div>
</div>
`;

exports[`<MediaCard /> should renders correct <MediaCard /> with correct videoSrc prop 1`] = `
<div
  className="text-center md:text-left"
>
  <figure
    className="relative mb-3 md:mb-4"
    style={
      Object {
        "height": "420px",
      }
    }
  >
    <div
      className="h-full overflow-hidden relative md:rounded-lg"
    >
      <VideoPlayer
        autoplay={false}
        className="object-cover absolute inset-0 w-full h-full"
        controls={false}
        loop={true}
        muted={true}
        playerRef={
          Object {
            "current": null,
          }
        }
        playsInline={true}
        poster="image.jpg"
        url="video.avi"
      />
    </div>
    <button
      className="absolute inset-0 w-full h-full"
      onClick={[Function]}
      style={
        Object {
          "background": "linear-gradient(0deg, rgba(0, 0, 0, 0.75) 0%, rgba(0, 0, 0, 0.15) 50%)",
        }
      }
    />
  </figure>
  <h3
    className="px-3 mb-2 font-serif text-2xl font-medium md:px-0"
  >
    Test title
  </h3>
  <div
    className="px-3 max-w-sm md:px-0"
  >
    Test description
  </div>
</div>
`;
